app/src/main/java/org/p2p/wallet/striga/signup/ui/StrigaSignUpFirstStepFragment.kt

```diff
@@ -0,0 +1,194 @@
+import org.p2p.wallet.intercom.IntercomService
+import org.p2p.wallet.striga.signup.StrigaSignUpFirstStepContract
+import org.p2p.wallet.striga.signup.model.StrigaSignupFieldState
+import org.p2p.wallet.striga.signup.repository.model.StrigaSignupDataType
+import org.p2p.wallet.striga.countrypicker.StrigaCountryPickerFragment
+import org.p2p.wallet.utils.getParcelableCompat
+import org.p2p.wallet.utils.popBackStack
+import org.p2p.wallet.utils.replaceFragment
+import org.p2p.wallet.utils.replaceFragmentForResult
+import org.p2p.wallet.utils.toDp
+import org.p2p.wallet.utils.viewbinding.getDrawable
+import org.p2p.wallet.utils.viewbinding.viewBinding
+
+class StrigaSignUpFirstStepFragment :
+    BaseMvpFragment<StrigaSignUpFirstStepContract.View, StrigaSignUpFirstStepContract.Presenter>(
+        R.layout.fragment_striga_sign_up_first_step
+    ),
+    StrigaSignUpFirstStepContract.View {
+
+    companion object {
+        const val REQUEST_KEY = "REQUEST_KEY"
+        const val RESULT_KEY = "RESULT_KEY"
+        fun create() = StrigaSignUpFirstStepFragment()
+    }
+
+    override val presenter: StrigaSignUpFirstStepContract.Presenter by inject()
+    private val binding: FragmentStrigaSignUpFirstStepBinding by viewBinding()
+
+    private lateinit var editTextFieldsMap: Map<StrigaSignupDataType, UiKitEditText>
+
+    private var birthdayMaskFormatter = SimpleMaskFormatter("##.##.####")
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        editTextFieldsMap = createEditTextsMap()
+        super.onViewCreated(view, savedInstanceState)
+        requireActivity().onBackPressedDispatcher.addCallback(viewLifecycleOwner) {
+            onBackPressed()
+        }
+        with(binding) {
+            uiKitToolbar.setNavigationOnClickListener { onBackPressed() }
+            uiKitToolbar.setOnMenuItemClickListener {
+                if (it.itemId == R.id.helpItem) {
+                    view.hideKeyboard()
+                    IntercomService.showMessenger()
+                    return@setOnMenuItemClickListener true
+                }
+                false
+            }
+
+            StrigaSignupDataType.cachedValues.forEach { dataType ->
+                val inputView = editTextFieldsMap[dataType] ?: return@forEach
+                inputView.addOnTextChangedListener { editable ->
+                    presenter.onFieldChanged(newValue = editable.toString(), type = dataType)
+                }
+            }
+            editTextCountry.setOnClickListener {
+                presenter.onCountryClicked()
+            }
+            buttonNext.setOnClickListener {
+                presenter.onSubmit()
+            }
+
+            binding.editTextBirthday.input.addTextChangedListener(
+                birthdayMaskFormatter.textWatcher(
+                    binding.editTextBirthday.input
+                )
+            )
+        }
+    }
+
+    override fun onStop() {
+        super.onStop()
+        presenter.onStop()
```

Comment:
could you rename the method in presenter and give the name of it's real action

Reply:
🆗